### PR TITLE
drivers/isl29020 : Expose Configurations to Kconfig 

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in ISL29020 Sensor Device driver to Kconfig.

### Testing procedure
1. New documentation was built using doxygen 

The build worked fine.

2. New macro was introduced in tests/driver_tmp00x/main.c for testing.

```
#define STR(x)   #x
#define SHOW_DEFINE(x) printf("%s=%s\n", #x, STR(x))
```
Firmware was uploaded to FIT IoT Lab test bed.

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-311-gec19c-Kconfig_isl29020_tests)
CONFIG_ISL29020_DEFAULT_ADDRESS=0x44
ISL29020 light sensor test application

Initializing ISL29020 sensor
[Failed]

#### Usage with CFLAGS 

/tests/driver_isl29020/Makefile

> CFLAGS += -DCONFIG_ISL29020_DEFAULT_ADDRESS=0x45

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-311-gec19c-Kconfig_isl29020_tests)
CONFIG_ISL29020_DEFAULT_ADDRESS=0x45
ISL29020 light sensor test application

Initializing ISL29020 sensor
[Failed]

#### Usage with Kconfig

/tests/driver_isl29020/

> make menuconfig

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-312-gc0d85b-Kconfig_isl29020_tests)
CONFIG_ISL29020_DEFAULT_ADDRESS=0x45
ISL29020 light sensor test application

Initializing ISL29020 sensor
[Failed]

Note : The sensor is not available for interfacing hence configurability of macros were only tested.

### Issues/PRs references

#12888
@leandrolanzieri  Please merge this only after  #13981

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
Please rebase

**Comment by akshaim** [comment]
> Please rebase

Done.

File: drivers/include/isl29020.h [hunk 1]
```diff
@@ -39,13 +39,13 @@ extern "C" {
 /**
  * @brief Default address
  *
- * The address depends on the status of A0 Pin.
- * Default address corresponds to A0 connected to GND.
- * For more information on SerialBus Address, refer Section I2C
- * Interface on Page 3 of datasheet.
+ * The address depends on the status of A0 Pin. Default
```
Comment (leandrolanzieri) [inline drivers/include/isl29020.h:42]: Please only wrap lines to adjust to 80 chars.

Comment (akshaim) [reply drivers/include/isl29020.h]: Done.

**Comment by leandrolanzieri** [comment]
@akshaim please squash and rebase

**Comment by akshaim** [comment]
> @akshaim please squash and rebase

Done. Thanks for the review

**Pull request merged**